### PR TITLE
Add isLeapYear, daysInMonth, clampDayOfMonth

### PR DESCRIPTION
**Pull request by cbracken**
This change extracts these functions to a new file, util.dart, and makes
them public.

### CODE REVIEW

**Comment by cbracken** [comment]
Fixes #378.

File: lib/src/time/util.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+part of quiver.time;
+
+/// Days in a month. This array uses 1-based month numbers, i.e. January is
+/// the 1-st element in the array, not the 0-th.
+const _DAYS_IN_MONTH = const [
```
Comment (yjbanov) [inline lib/src/time/util.dart]: lowerCamelCase

File: lib/src/time/util.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  30,
+  31,
+  31,
+  30,
+  31,
+  30,
+  31
+];
+
+int daysInMonth(int year, int month) =>
```
Comment (yjbanov) [inline lib/src/time/util.dart]: dartdocs

File: lib/src/time/util.dart [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    (month == DateTime.FEBRUARY && isLeapYear(year))
+        ? 29
+        : _DAYS_IN_MONTH[month];
+
+bool isLeapYear(int year) =>
```
Comment (yjbanov) [inline lib/src/time/util.dart]: dartdocs

File: lib/src/time/util.dart [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+///
+/// February 31, 2013 => February 28, 2013
+///
+/// When jumping from month to month or from leap year to common year we may
+/// end up in a month that has fewer days than the month we are jumping from.
+/// In that case it is impossible to preserve the exact date. So we "clamp" the
+/// date value to fit within the month. For example, jumping from March 31 one
+/// month back takes us to February 28 (or 29 during a leap year), as February
+/// doesn't have 31-st date.
+int clampDate(int date, int year, int month) =>
```
Comment (yjbanov) [inline lib/src/time/util.dart]: Is this a good name? `clampDateToMonth`?

Comment (yjbanov) [inline lib/src/time/util.dart]: Should parameters be named? I hear `DateTime` is changing its signature to name _its_ parameters.

**Comment by yjbanov** [comment]
Also, what do you think of moving these into a new library named `calendar`? It's weird from the discoverability standpoint to put them in `time.dart`.

**Comment by yjbanov** [comment]
lgtm

File: lib/src/time/util.dart (hunk 4 above)
Comment (cbracken) [reply lib/src/time/util.dart]: Done - looks like we almost certainly depend on `meta` via `matcher` anyway` :)

Comment (cbracken) [reply lib/src/time/util.dart]: Done: `clampDayOfMonth`

**Pull request merged**